src/pdftoraw.c

```diff
@@ -0,0 +1,189 @@
+    height = DPI * height / 72.0;
+    w = (int)ceil(width);
+    h = (int)ceil(height);
+    s = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, w, h);
+    cr = cairo_create(s);
+    cairo_scale (cr, DPI/72.0, DPI/72.0);
+    cairo_save (cr);
+    poppler_page_render_for_printing(page, cr);
+    cairo_restore (cr);
+    pix = gdk_pixbuf_get_from_surface(s, 0, 0, w, h);
+    g_object_unref(page);
+    data = gdk_pixbuf_get_pixels (pix);
+    fwrite(data, 1, (w * h * 4), stdout);
+    return data;
+}
+
+static int
+pdf_page_width_get(PopplerDocument *doc,
+                   int page_nr)
+{
+    double width = 0;
+    int w = 0;
+
+    PopplerPage *page = poppler_document_get_page(doc, page_nr);
+    poppler_page_get_size(page, &width, NULL);
+    width = DPI * width / 72.0;
+    w = (int)ceil(width);
+    g_object_unref(page);
+    return w;
+}
+
+static int
+pdf_page_height_get(PopplerDocument *doc,
+                    int page_nr)
+{
+    double height = 0;
+    int h = 0;
+
+    PopplerPage *page = poppler_document_get_page(doc, page_nr);
+    poppler_page_get_size(page, NULL, &height);
+    height = DPI * height / 72.0;
+    h = (int)ceil(height);
+    g_object_unref(page);
+    return h;
+}
+
+int
+main(int argc, char **argv)
+{
+    PopplerDocument *doc = NULL;
+    GError *err = NULL;
+    GFile *in = NULL;
+    char *filename = NULL;
+    int long_index =0;
+    int page = 0;
+    int opt= 0;
+    int test = 0, raw = 0, npages = 0, width = 0, height = 0;
+
+    while ((opt = getopt_long(argc, argv,"f:tpw:W:r:h",
+                   long_options, &long_index )) != -1) {
+        switch (opt) {
+             case 'f' :
+                 filename = strdup(optarg);
+                 break;
+             case 't' :
+                 test = 1;
+                 break;
+             case 'p' :
+                 npages = 1;
+                 break;
+             case 'w' : page = atoi(optarg);
+                 width = 1;
+                 break;
+             case 'W' : page = atoi(optarg);
+                 height = 1;
+                 break;
+             case 'r' : page = atoi(optarg);
+                 raw = 1;
+                 break;
+             default: print_usage();
+                 exit(EXIT_FAILURE);
+        }
+    }
+    if (filename == NULL) {
+        print_usage();
+        exit(EXIT_FAILURE);
+    }
+    in = g_file_new_for_path(filename);
+    doc = poppler_document_new_from_gfile(in, NULL, NULL, &err);
+    if (err)
+    {
+       g_error_free(err);
+       g_object_unref(in);
+       exit(EXIT_FAILURE);
+    }
+
+    if (test == 1) {
+        printf("1");
+    }
+    else if (raw == 1) {
+        pdf_page_raw_get(doc, page);
+    }
+    else if (npages == 1) {
+        printf("%d", pdf_number_pages_get(doc));
+    }
+    else if (width == 1) {
+        printf("%d", pdf_page_width_get(doc, page));
+    }
+    else if (height == 1) {
+        printf("%d", pdf_page_height_get(doc, page));
+    }
+    else {
```

Comment:
Formatting nit, indent braces in GNU style, like:

```c
if (foo)
  {
    ...
  }
else
  {
    ...
  }
```

Reply:
fixed 3a6cd1e